### PR TITLE
feat(servicestage/env): add new resource to manage environments

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new resource to manage environments and according to the ver.3 APIs.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to manage environments
```

## PR Checklist

* [x] Tests added/passed.

```
../coverage.sh -o servicestage -f TestAccV3Environment_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/servicestage" -v -coverprofile="./huaweicloud/services/acceptance/servicestage_coverage.cov" -coverpkg="./huaweicloud/services/servicestage" -run TestAccV3Environment_basic -timeout 360m -parallel 10
=== RUN   TestAccV3Environment_basic
=== PAUSE TestAccV3Environment_basic
=== CONT  TestAccV3Environment_basic
--- PASS: TestAccV3Environment_basic (112.28s)
PASS
coverage: 9.2% of statements in ./huaweicloud/services/servicestage
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/servicestage      112.327s        coverage: 9.2% of statements in ./huaweicloud/services/servicestage
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    image

### CODE REVIEW

**Comment by Lance52259** [comment]
```
 make testacc TEST='./huaweicloud/services/acceptance/servicestage' TESTARGS='-run=TestAccV3Environment_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/servicestage -v -run=TestAccV3Environment_basic -timeout 360m -parallel 4
=== RUN   TestAccV3Environment_basic
=== PAUSE TestAccV3Environment_basic
=== CONT  TestAccV3Environment_basic
--- PASS: TestAccV3Environment_basic (110.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/servicestage      110.947s
```

File: huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_environment_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/servicestage"
+)
+
+func getV3EnvFunc(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := conf.ServiceStageV2Client(acceptance.HW_REGION_NAME)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_environment_test.go]:
```suggestion
	client, err := conf.ServiceStageV1Client(acceptance.HW_REGION_NAME)
```
Try to be consistent with the resource code.

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestagev3_environment.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+		return diag.Errorf("error creating environment: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	envId, err := jmespath.Search("id", respBody)
+	if err != nil {
```
Comment (deer-hang) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestagev3_environment.go]:
```suggestion
	if err != nil || envId == nil {
```

File: huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_environment_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_environment_test.go]: Should be NewServiceClient

**Pull request merged**